src/bun.js/webcore/encoding.zig

```diff
@@ -676,25 +676,56 @@ pub const TextDecoder = struct {
         globalThis: *JSC.JSGlobalObject,
         callframe: *JSC.CallFrame,
     ) callconv(.C) ?*TextDecoder {
-        var args_ = callframe.arguments(1);
+        var args_ = callframe.arguments(2);
         var arguments: []const JSC.JSValue = args_.ptr[0..args_.len];
 
-        var encoding = EncodingLabel.@"UTF-8";
-        if (arguments.len > 0) {
-            if (!arguments[0].isString()) {
-                globalThis.throwInvalidArguments("TextDecoder(encoding) label is invalid", .{});
-                return null;
-            }
+        var decoder = getAllocator(globalThis).create(TextDecoder) catch unreachable;
```

Comment:
If we validate arguments and then allocate, we can avoid deallocation

It would also be okay to create a stack-local instance of TextDecoder and then allocate and assign it at the end